internal/services/synapse/synapse_pipeline_resource_test.go

```diff
@@ -0,0 +1,357 @@
+		data.ImportStep(),
+	})
+}
+
+func TestAccSynapsePipeline_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_pipeline", "test")
+	r := PipelineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.update1(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update2(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSynapsePipeline_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_pipeline", "test")
+	r := PipelineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSynapsePipeline_activities(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_pipeline", "test")
+	r := PipelineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.activities(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.activitiesUpdated(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.activities(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSynapsePipeline_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_synapse_pipeline", "test")
+	r := PipelineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (t PipelineResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.PipelineID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	environment := client.Account.Environment
+	pipelinesClient, err := client.Synapse.PipelinesClient(id.WorkspaceName, environment.SynapseEndpointSuffix)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := pipelinesClient.GetPipeline(ctx, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r PipelineResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+	%s
+
+resource "azurerm_synapse_pipeline" "test" {
+  name                 = "acctest%d"
```

Comment:
Can we add some capitalized character in the name? 
```suggestion
  name                 = "acctest-SynapsePipeline%d"
```

Reply:
Fixed